homeassistant/components/zwave_js/update.py

```
@@ -0,0 +1,199 @@
+        """Process updates."""
+        if self.available_firmware_updates:
+            self._latest_version_firmware = firmware = get_latest_firmware(
+                self.available_firmware_updates
+            )
+            self._attr_latest_version = firmware.version
+        else:
+            self._latest_version_firmware = None
+            self._attr_latest_version = self._attr_installed_version
+        if write_state:
+            self.async_write_ha_state()
+
+    async def async_release_notes(self) -> str | None:
+        """Get release notes."""
+        if self._latest_version_firmware is None:
+            return None
+        return self._latest_version_firmware.changelog
+
+    async def async_install(
+        self, version: str | None, backup: bool, **kwargs: Any
+    ) -> None:
+        """Install an update."""
+        if version is None and self._latest_version_firmware:
+            firmware = self._latest_version_firmware
+        else:
+            try:
+                firmware = next(
+                    firmware
+                    for firmware in self.available_firmware_updates
+                    if firmware.version == version
+                )
+            except StopIteration as err:
+                raise ValueError(f"Version {version} not found") from err
+        self._attr_in_progress = True
+        try:
+            for file in firmware.files:
+                await self.driver.controller.async_begin_ota_firmware_update(
+                    self.node, file
+                )
+        except BaseZwaveJSServerError as err:
+            raise HomeAssistantError(err) from err
+        else:
+            self._attr_installed_version = firmware.version
+            self.available_firmware_updates.remove(firmware)
+            self._async_process_updates()
+        finally:
+            self._attr_in_progress = False
+
+    async def async_poll_value(self, _: bool) -> None:
+        """Poll a value."""
+        LOGGER.error(
```

Should we raise an error ?

---

we don't anywhere else (node status sensor, ping button). If we change the behavior here we should change it there as well

---

ok not for this PR.